src/librustc_middle/hir/map/collector.rs

```diff
@@ -232,38 +232,13 @@ impl<'a, 'hir> NodeCollector<'a, 'hir> {
         }
     }
 
-    fn insert(&mut self, span: Span, hir_id: HirId, node: Node<'hir>) {
-        self.insert_with_hash(span, hir_id, node, Fingerprint::ZERO)
+    fn insert(&mut self, hir_id: HirId, node: Node<'hir>) {
+        self.insert_with_hash(hir_id, node, Fingerprint::ZERO)
     }
 
-    fn insert_with_hash(&mut self, span: Span, hir_id: HirId, node: Node<'hir>, hash: Fingerprint) {
+    fn insert_with_hash(&mut self, hir_id: HirId, node: Node<'hir>, hash: Fingerprint) {
         let entry = Entry { parent: self.parent_node, node };
 
-        // Make sure that the DepNode of some node coincides with the HirId
-        // owner of that node.
-        if cfg!(debug_assertions) {
-            let node_id = self.definitions.hir_id_to_node_id(hir_id);
-
-            if hir_id.owner != self.current_dep_node_owner {
-                let node_str = match self.definitions.opt_local_def_id(node_id) {
-                    Some(def_id) => self.definitions.def_path(def_id).to_string_no_crate(),
-                    None => format!("{:?}", node),
-                };
-
-                span_bug!(
-                    span,
-                    "inconsistent DepNode at `{:?}` for `{}`: \
-                     current_dep_node_owner={} ({:?}), hir_id.owner={} ({:?})",
-                    self.source_map.span_to_string(span),
-                    node_str,
-                    self.definitions.def_path(self.current_dep_node_owner).to_string_no_crate(),
-                    self.current_dep_node_owner,
-                    self.definitions.def_path(hir_id.owner).to_string_no_crate(),
-                    hir_id.owner,
-                )
-            }
-        }
```

Comment:
Is it possible to keep this check by moving it to AST lowering, for example?
IIRC, it caught a number of issues in the past.

Reply:
I'll take a look

Reply:
I wrongly removed it because i did not realized that the conversion `HirId` -> `NodeId` -> `LocalDefId` is actually the same conversion as `HirId` -> `LocalDefId` with `opt_hir_id_to_local_def_id`.